presto-parser/src/main/java/com/facebook/presto/sql/tree/Window.java

```diff
@@ -13,99 +13,21 @@
  */
 package com.facebook.presto.sql.tree;
 
-import com.google.common.collect.ImmutableList;
-
-import java.util.List;
-import java.util.Objects;
 import java.util.Optional;
 
-import static com.google.common.base.MoreObjects.toStringHelper;
-import static java.util.Objects.requireNonNull;
-
-public class Window
+public abstract class Window
```

Comment:
Actually, I'd probably also add `getSpecification()` to the base class, then `WindowName` would resolve specification through the `WindowDefinition` it'd hold the reference to.
Otherwise I don't see a point of having a common super class for `WindowName` and `WindowInline` because in any real case we have to deal with them separately and differently by resolving `WindowSpecification` differently.

Against my proposal, I'd find the fact that `WindowInline` and `WindowName` comes from AST so it's sketchy to manipulate structures which are meant to come directly from parser.